usb/core.py

```diff
@@ -671,9 +671,11 @@ def __getitem__(self, index):
 
 
     def _str(self):
+        power_multiplier = _lu.MAX_POWER_UNITS_USB_SUPERSPEED if ((self.device.bcdUSB & 0xff00)>>8) == 3 else _lu.MAX_POWER_UNITS_USB2p0
```

Comment:
I think we should first check `self.device.speed` (from [`libusb_get_device_speed`](https://libusb.sourceforge.io/api-1.0/group__libusb__dev.html#ga58c4e448ecd5cd4782f2b896ec40b22b)), and only fallback to `bcdUSB` when that isn't available.

Also, I suggest replacing the mask and exact comparison with `self.device.bcdUSB >= 0x0300`.

Finally, the choice of multiplier is duplicated between this function and `_get_full_descriptor_str`, and non trivial; can you make sure it's only done in a single place?

Reply:
Thanks for your feedback. I've pushed up a commit which addresses your comments - let me know what you think.